test-types/jasmine/expectAsync.ts

```diff
@@ -0,0 +1,7 @@
+declare const promise: Promise<object>
+
+it('should do something', () => {
```

Comment:
Let's add test for valid usage of expect async as well

Reply:
Dome, but I've added a test for `then` instead. `await`ing a non-promise value won't make TS issue an error, only a suggestion from the language service in the editor.